core-models/src/main/scala/com/mesosphere/usi/core/models/FetchUri.scala

```diff
@@ -0,0 +1,36 @@
+package com.mesosphere.usi.core.models
+
+import java.net.URI
+
+/**
+  * Specification used by the [Mesos fetcher](http://mesos.apache.org/documentation/latest/fetcher/) to download
+  * resources into the sandbox directory of a task in preparation of running the task. Defaults taken from Mesos
+  * proto files.
+  *
+  * @param uri artifact URI to download
+  * @param extract if true, the artifact will be treated as an archive an extracted to the sandbox. See the list
+  *                of the supported archive types below
+  * @param executable if true, the fetch result will be changed to be executable (by “chmod”) for every user
+  * @param cache if true, the fetcher cache is to be used for the URI
+  * @param outputFile if set, the fetcher will use that name for the copy stored in the sandbox directory
+  */
+case class FetchUri(
+    uri: URI,
```

Comment:
why not akka.http.Uri? it's more strict in following the RFC 3986.

Reply:
because it's Java-friendly. also, where exactly it breaks the RFC?